### DISCUSSION TITLE
Docker: how to use seqcli in a running container?.

### DISCUSSION THREAD

**Discussion by rj-au** [datalust/seq-tickets#2293] [Help] [answered]
I'd like to run the seqcli in the running Seq server container (for local development). My O/S is Linux.

Why: So I don't have to set `--server` option, and attach to a specific docker private network, when using the seqcli image. I need to do this when I want to manually import a JSON log file (downloaded from GH-Actions artifacts) from an API used for E2Es when there are issues. The API runs as a container via Github actions. I could ship to our production Seq instance, I don't want to mix Prod (and lower environment) logs with logs from tests.

Running seqcli from the image works fine:
```bash
docker run --rm datalust/seq:latest seqcli
# Usage: seq <command> [<args>]
```

But from the running from the running Seq server container errors out:
```bash
docker exec -it local-seq seqcli
# The application to execute does not exist: '/usr/bin/seqcli.dll'.
```

Any idea why I get this error?

**Comment by KodrAus** [comment, marked as answer]
Hi @richard-stafflink :wave: The executable lives under `/seqsvr/Client/seqcli`. It looks like there might be an issue with how we link `seqcli` to `/bin/seqcli` in the container that makes this not work. If you run:

```
docker exec -it local-seq /seqsvr/Client/seqcli
```

it should work:

```
Usage: seqcli <command> [<args>]
Type `seqcli help` for available commands
```

**Reply by rj-au** [reply]
Nice! thank you, that works.

**Comment by koide** [comment]
I ran into the same problem and it seems that for some reason you have put an actual copy of the `seqcli` binary on `/usr/bin` instead of a link and that's missing dependencies and not working, giving this error: `The application to execute does not exist: '/usr/bin/seqcli.dll'.`

If on the container image I run `ln -sf /seqsvr/Client/seqcli /usr/bin/seqcli` then running `seqcli` directly works.

**Reply by nblumhardt** [reply]
Thanks @koide, we'll raise this as a bug and sort it out 👍
